hal/src/rtl872x/backup_ram_hal.cpp

```diff
@@ -116,14 +139,17 @@ int hal_backup_ram_sync(void* reserved) {
     BackupRamLock lk;
     if (memcmp((void*)&platform_backup_ram_all_start, backupRamShadow, sizeof(backupRamShadow))) {
         memcpy(backupRamShadow, (void*)&platform_backup_ram_all_start, sizeof(backupRamShadow));
-        if (SYSTEM_FLAG(restore_backup_ram) != 1) {
-            SYSTEM_FLAG(restore_backup_ram) = 1;
+        if (!(SYSTEM_FLAG(restore_backup_ram) & SYSTEM_FLAG_RESTOR_BACKUP_RAM_MASK)) {
+            SYSTEM_FLAG(restore_backup_ram) |= SYSTEM_FLAG_RESTOR_BACKUP_RAM_MASK;
             dct_write_app_data(&system_flags, DCT_SYSTEM_FLAGS_OFFSET, DCT_SYSTEM_FLAGS_SIZE);
```

Comment:
We store `system_flags` to DCT in addition to keeping it in retained SRAM in case we lose power and want to restore some retained values from flash?

Reply:
Yes.